### PR TITLE
Tweaks for alpha release

### PR DESCRIPTION
**Pull request by neatudarius**
Tweaks for alpha release (Saint Louis demo)

Related issues: #12 #11

Updates:
* tweaks in docs
* add more toolchain presets for local testing
* add script helpers for easier run of all configurations on local setup
* set git attribute for better repo stats (exclude latex/paper files from stats): check old vs new stats
image
image

* * add script helper for local linters. TBD integration with CI. 
    * Proposed linters: `C++: clang-format`,  `Shell: shellcheck`, `MD: markdownlint`, `YAML: yamlfmt`, `Cmake: cmake-format`
   * Check usages in `scripts/README.md` for scripts and linters.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*.bib -linguist-detectable
```
[the remark this answers is not in the document]
Comment (neatudarius) [reply .gitattributes:1]: yes, please check screen shots attached in PR description.

File: README.md [hunk 2]
```diff
@@ -4,27 +4,18 @@
...
+This repository implements `std::optional` extensions targeting C++26. The `Beman.Optional26` library aims to evaluate the stability, the usability, and the performance of these proposed changes before they are officially adopted by WG21 into the C++ Working Draft. Additionally, it allows developers to use these new features before they are implemented in major standard library compilers.
 
 **Implements**:
 * [Give *std::optional* Range Support (P3168R1)](https://wg21.link/P3168R1)
 * [`std::optional<T&>` (P2988R5)](https://wg21.link/P2988R5)
 
 ## License
 
-Source is licensed with the Apache 2.0 license with LLVM exceptions
+Source, documentation and associated papers are licensed with the Apache 2.0 license with LLVM exceptions.
```
[the remark this answers is not in the document]
Comment (neatudarius) [reply README.md:16]: I received this suggestion from @camio . Let's discuss here. I don't have any preferences.

**Comment by steve-downey** [comment]
For every complex problem there is an answer that is clear, simple, and
wrong. --- H.L Mencken

This is probably not the right place to sort out license details.
I agree that we need to make the licensing as single as possible, but no
simpler.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (camio) [reply README.md:16]: My thought was to unify the licenses to simplify things for readers. The Carbon project seems to use Apache2+LLVM licensing for their documentation too. See https://github.com/carbon-language/carbon-lang/blob/trunk/README.md?plain=1 for an example.

[the remark this answers is not in the document]
Comment (camio) [reply README.md:16]: I'm okay with having the docs under a different documentation license.

[the remark this answers is not in the document]
Comment (neatudarius) [reply README.md]: Cool. Then I temporary reverted the changes related to the `License` subsection. Let's move the discussion in private and continue with the review in this PR.

File: CMakePresets.json [hunk 3]
```diff
@@ -31,12 +31,33 @@
       "description": "Build with GCC 14 compilers",
       "toolchainFile": "${sourceDir}/etc/gcc-14-toolchain.cmake"
     },
+    {
+      "name": "gcc-13",
```
Comment (neatudarius) [inline CMakePresets.json:35]: Useful for local debug.

File: scripts/test-all.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# Expected to be run from the root of the project.
+# Expected all configurations to be present in the CMakeLists.txt file and installed on the system.
+# Usage: ./scripts/test-all.sh
+
+set -e
+set -x
+
+rm -rf .build
+
+cmake --workflow --preset gcc-14   &> /dev/null && echo "passed" || echo "gcc-14 test failed."
```
Comment (neatudarius) [inline scripts/test-all.sh:15]: Useful for local testing (try to mirror current flows from CI).

File: .markdownlint.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "MD013": false
```
Comment (neatudarius) [inline .markdownlint.json:2]: Ignore line length.

**Review by steve-downey — APPROVED** [review]
Looks good to me. 
And hopefully to others tonight!

**Comment by neatudarius** [comment]
Link to #14 - as we introduce P3168R2

**Pull request merged**